validation/src/test/scala/hmda/validation/rules/ts/validity/V155Spec.scala

```diff
@@ -12,15 +12,17 @@ class V155Spec extends PropSpec with PropertyChecks with MustMatchers with TsGen
   property("A valid email address for the institution must be provided") {
     forAll(tsGen) { ts =>
       whenever(ts.contact.email != "") {
-        V155(ts.contact) mustBe Success()
+        V155(ts) mustBe Success()
       }
     }
   }
 
   property("Invalid emails should be detected") {
-    forAll(badContactGen) { contact =>
-      whenever(contact.email != "") {
-        V155(contact) mustBe Failure("is not a valid email")
+    forAll(tsGen) { ts =>
+      val badContact = badContactGen.sample.getOrElse(Contact("", "", "", ""))
+      val badTs = ts.copy(contact = badContact)
+      whenever(badTs.contact.email != "") {
+        V155(badTs) mustBe Failure("is not a valid email")
```

Comment:
This `whenever
Shouldn't the same failure also happen when the contact email *is* the empty string? It seems worth covering that case too. (I could see wanting to put it into a separate test, but I don't see that test in this file.)